### PR TITLE
fix(tooltip): Open hovered tooltip while pointer is moving

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #6278 #6615 #6785

## Summary

- Only clears the hovering timeout for toggling a tooltip when:
  - An active tooltip is being hovered.
  - An active tooltip exists and a tooltip reference element is not being hovered.
- This means a newly hovered tooltip can be opened without having to stop moving the mouse.
- Had to make some adjustments for clicked tooltips when `closeOnClick` is true.
- Cleanup some code 
- Adds test

### CODE REVIEW

File: src/components/tooltip/TooltipManager.ts [hunk 1]
```diff
@@ -172,21 +174,26 @@ export default class TooltipManager {
...
-    if (!tooltip || tooltip === this.clickedTooltip) {
-      this.clickedTooltip = null;
+    if (!tooltip || this.isCloseOnClickTooltip(tooltip)) {
       return;
     }
 
     this.toggleFocusedTooltip(tooltip, value);
   }
+
+  private isCloseOnClickTooltip(tooltip: HTMLCalciteTooltipElement): boolean {
```
Comment (jcfranco) [inline src/components/tooltip/TooltipManager.ts:196]: Can this be renamed to also convey that it's checking for the clicked tooltip?

File: src/components/tooltip/tooltip.e2e.ts [hunk 2]
```diff
@@ -717,4 +717,55 @@ describe("calcite-tooltip", () => {
...
+      },
+      {
+        delay: TOOLTIP_DELAY_MS + TOOLTIP_DELAY_MS * 0.5,
+        value: true
+      }
+    ];
+
+    const hoverPromises: Promise<void>[] = pointerMoves.map(async ({ delay }) => {
+      await page.waitForTimeout(delay);
+      await page.$eval("#ref", (elm: HTMLElement) => {
```
Comment (jcfranco) [inline src/components/tooltip/tooltip.e2e.ts:760]: Nitpick: For consistency, `elm` ➡ `el`?

**Review by jcfranco — APPROVED** [review]
🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯
🗯↖🗯🗯🗯↖🗯↖🗯🗯🗯↖🗯↖🗯🗯🗯↖🗯
🗯↖↖🗯↖↖🗯↖🗯🗯🗯↖🗯🗯↖🗯↖🗯🗯
🗯↖🗯↖🗯↖🗯↖🗯🗯🗯↖🗯🗯🗯↖🗯🗯🗯
🗯↖🗯🗯🗯↖🗯↖🗯🗯🗯↖🗯🗯🗯↖🗯🗯🗯
🗯↖🗯🗯🗯↖🗯🗯↖↖↖🗯🗯🗯🗯↖🗯🗯🗯
🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯


🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯
🗯↖🗯🗯🗯↖🗯↖↖↖🗯🗯↖↖↖🗯↖↖↖↖🗯↖🗯
🗯↖↖🗯🗯↖🗯🗯↖🗯🗯↖🗯🗯🗯🗯↖🗯🗯🗯🗯↖🗯
🗯↖🗯↖🗯↖🗯🗯↖🗯🗯↖🗯🗯🗯🗯↖↖↖🗯🗯↖🗯
🗯↖🗯🗯↖↖🗯🗯↖🗯🗯↖🗯🗯🗯🗯↖🗯🗯🗯🗯🗯🗯
🗯↖🗯🗯🗯↖🗯↖↖↖🗯🗯↖↖↖🗯↖↖↖↖🗯↖🗯
🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯🗯

**Comment by geospatialem** [comment]
@driskull Did we want to address the cursor movement outside of the reference element as it relates to https://github.com/Esri/calcite-components/issues/6785? 

The `tooltip` has improved for its visibility once the cursor moves away, however it still stays open when moving outside of the reference element and still moving the mouse.

tooltip-invoke-verify

**Comment by driskull** [comment]
@geospatialem if you hover over the tooltip it will reset the timeout to hide it.

**Comment by driskull** [comment]
@geospatialem we can use that issue or a new one. Whichever is easier.

**Comment by geospatialem** [comment]
Let's use the original for tracking - will assign it back to 'ya. Thanks Matt!

**Pull request merged**